packages/mds-stream-processor/index.ts

```diff
@@ -17,45 +17,78 @@
 import logger from '@mds-core/mds-logger'
 import { SingleOrArray } from '@mds-core/mds-types'
 import { asArray } from '@mds-core/mds-utils'
-import { StreamProcessorController, StreamSink, StreamSource, StreamTransform } from './@types'
+import { DeadLetterSink, StreamProcessorController, StreamSink, StreamSource, StreamTransform } from './@types'
 
 // StreamProcessor - Read from source, apply transform to each message, and write to sink
 export const StreamProcessor = <TMessageIn, TMessageOut>(
   source: StreamSource<TMessageIn>,
   transform: StreamTransform<TMessageIn, TMessageOut>,
-  sinks: SingleOrArray<StreamSink<TMessageOut>>
+  sinks: SingleOrArray<StreamSink<TMessageOut>>,
```

Comment:
I've often thought about modifying this signature to something like:
```ts
export const StreamProcessor = <TMessageIn, TMessageOut>(
  options: Partial<{ deadLetterSinks: SingleOrArray<DeadLetterSink<TMessageIn>> }> = {}
) => (
  source: StreamSource<TMessageIn>,
  transform: StreamTransform<TMessageIn, TMessageOut>,
  sinks: ...StreamSink<TMessageOut>[]
): StreamProcessorControlller => { ... }
```
Allow expansion of optional options by currying and changing the `sinks` since there's usually only one to a rest parameter. Calling would be something like:
```ts
StreamProcessor()(mySource, myTransform, mySink1, mySink2)
```
Not sure if that's better or not, but I've often contemplated making that change.

Reply:
Mmm yeah, that could be interesting. I wonder how that'd work out with having dead letter sinks now though, were you thinking that we'd do a rest parameter for the dead letter sinks instead?

Reply:
Yeah, that's definitely an interesting idea. I'll see how using the current signature in practice is, and then adjust it to something along these if it feels clunky :)